tests/optimize/test_objective.py

```diff
@@ -0,0 +1,30 @@
+"""Unit tests for objective module."""
+import unittest
+
+import relentless
+
+class QuadraticObjective(relentless.optimize.ObjectiveFunction):
+    """Mock objective function used to test relentless.optimize.ObjectiveFunction"""
+
+    def __init__(self, x):
+        self.x = x
+
+    def run(self):
+        val = self.x.value**2
+        grad = {self.x:2*self.x.value}
```

Comment:
To make these a little more interesting, how about you shift to make the function `(x-1)^2`? The minimum of this function is `x = 1`.